### DISCUSSION TITLE
Unable to connect postal SMTP server on secure connection

### DISCUSSION THREAD

**Discussion by airshell** [postalserver/postal#2683] [Help with using Postal]
Hi there, 

I have installed Postal 2.1.4 latest with Ubuntu 22.04 OS on my server and I am facing an issue to connect with postal SMTP server on secure connection from external web application.

I able to connect & send emails from external app with postal smtp server without secure connection, so it seems the issue is with postal not working with the SSL created by caddy server.

I have tried the solutions mentioned in earlier issues regarding similar problem and updated the config files **postal.yml** & **Caddyfile** as mentioned, I have copied the .cert & .key files inside as mentioned in issue #1572 :

https://github.com/postalserver/postal/discussions/1572#discussioncomment-2544886

Also updated the certificate & key path entry inside **postal.yml** file as mentioned in issue #1985 
https://github.com/postalserver/postal/issues/1985#issuecomment-1126271041

`smtp_server:
  # Specify configuration to the Postal SMTP server
  port: 25
  tls_enabled: true
  tls_certificate_path: config/mails.mydomain.com.cert
  tls_private_key_path: config/mails.mydomain.com.key`

But when I try to check if postal is using SSL with below command : 
`openssl s_client -connect mails.mydomain.com:25 -starttls smtp`

It returns no peer certificated available.

![no-certificate](https://github.com/postalserver/postal/assets/149798684/bd537b85-393a-492b-a41f-0e4ea585091d)

I have restarted my server as well as postal to see if that might be needed.

If anyone can tell me what I am missing, that would be really helpful.

Thanks
Sunny

**Comment by willpower232** [comment]
If you run `postal bash worker` that should put you inside the container, you could then `ls /config/mails.mydomain.com.key` to confirm the file is there and also `cat` or `less` it to confirm its contents are correct.

**Reply by airshell** [reply]
Hi Will,

Thanks for your reply, when I run` postal bash worker` it put me inside the container `postal@mails:~/app$`, 
then I use  `ls /config/`command which shows the file present in config directory of postal container:
Caddyfile  `mails.mydomain.crt`  `mails.mydomain.key`  `postal.yml ` `signing.key`

And I also checked the key content with `cat /config/mails.mydomain.com.key` which outputs the EC PRIVATE KEY for ssl certificate.

But when I try to check if postal is using this SSL with below command :
`openssl s_client -connect mails.mydomain.com:25 -starttls smtp`

It still shows no peer certificated available, is there anything else I can do to get this working.

**Reply by willpower232** [reply]
as long as the config file paths are /config/whatever and tls_enabled is true and you've restarted all of Postal then it should be working

**Reply by airshell** [reply]
I have tried restarting postal again, what should be the correct outpout of this command:  `openssl s_client -connect mails.mydomain.com:25 -starttls smtp`

This is what I see after restarting postal.

![Screenshot 2023-11-06 at 11 35 32 PM](https://github.com/postalserver/postal/assets/149798684/c08d02a5-8780-4d54-9177-1c7e915acae0)

**Reply by airshell** [reply]
@willpower232 do we need to rename the **mails.mydomain.cert** &  **mails.mydomain.key** files to **smtp.cert** and **smtp.key** like below after copying them to postal config directory:

`Copy the certificates to Postal's configuration directory`
```
    cp "$CERT_FILE" /opt/postal/config/smtp.crt
    cp "$KEY_FILE" /opt/postal/config/smtp.key
```

I saw it mentioned like this here -->  https://github.com/postalserver/postal/discussions/2673#discussioncomment-7382024

**Reply by willpower232** [reply]
it doesn't matter what the paths are named, as long as they are in the right order in the config file and link to valid files. Do you need to start the paths with a `/` or something?

**Reply by airshell** [reply]
Thanks for your reply, I really appreciate it.

I just mentioned this to highlight the **.key** & **.cert** file name difference not the path to the files.

I tried renaming both files to **smtp.key** & **smtp.cert** in config file **postal.yml** but that didn't helped in postal identifyig the SSL issued by caddy attaching image below.

![tls-files-path-postal-config](https://github.com/postalserver/postal/assets/149798684/abcee8e3-77ab-4231-88d1-497b497f87f5)

The postal server is able to send mails from SMTP without SSL authentication and if I choose SSL/TLS option in SMTP details on external web apps it starts showing connection time out error. 

Tested this on https://smtper.net/ as well, when I select "**Use Secured Connection**" checkbox the mail does not send and shows error.

**Reply by airshell** [reply]
I also tried adding "/" to file path in postal.yml file.

```
smtp_server:
  # Specify configuration to the Postal SMTP server
  port: 25
  tls_enabled: true
  tls_certificate_path: /config/smtp.cert
  tls_private_key_path: /config/smtp.key

```
![tls-files-path-postal](https://github.com/postalserver/postal/assets/149798684/09c77452-852e-4103-8f1f-bb5df9731192)

Stopped postal then restarted it again, but still getting error while using secure connection SSL/TLS.

**Reply by willpower232** [reply]
the only other thing I can think of is the permissions on the two files perhaps? otherwise I have no idea I'm afraid

**Reply by airshell** [reply]
Will can you have a look at this post, I believe this is permission related issue just like you said.

https://github.com/postalserver/postal/discussions/2689#discussioncomment-7513454
